### PR TITLE
Templating: removes old Angular variable system and featureToggle

### LINKED ISSUE #23219: Templating: remove legacy code and :confetti_ball:
:confetti_ball: :confetti_ball: :confetti_ball:

### PR DESCRIPTION
**Pull request by hugohaggmark**
### Don't merge this until we decided on when (after first patch release for 7.0?)

**What this PR does / why we need it**:
- Removes old variable Angular system and featuretoggle newVariables.
- Removes variableSrv
- Migrates RowOptions to React
- Makes the VariableModel interface less implicit

**Which issue(s) this PR fixes**:
Fixes #23219

**Special notes for your reviewer**:
@daniellee now that we removed the old Angular system these files need refactoring, I added skip to the tests for now until someone with better knowledge about the component can have a look.
`public/app/plugins/datasource/stackdriver/components/VariableQueryEditor.test.tsx`
`public/app/plugins/datasource/stackdriver/components/VariableQueryEditor.tsx`

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 1]
```diff
@@ -140,10 +142,10 @@ export class ShareSnapshot extends PureComponent<Props, State> {
     });
 
     // remove template queries
-    dash.getVariables().forEach(variable => {
+    dash.getVariables().forEach((variable: QueryVariableModel) => {
       variable.query = '';
-      variable.options = variable.current;
-      variable.refresh = false;
+      variable.options = variable.current ? [variable.current] : [];
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:147]: I need to verify this change

File: public/app/core/services/bridge_srv.ts [hunk 2]
```diff
@@ -84,7 +83,7 @@ export class BridgeSrv {
         if (changes) {
           const dash = getDashboardSrv().getCurrent();
           if (dash) {
-            this.variableSrv.templateVarsChangedInUrl(changes);
```
Comment (hugohaggmark) [inline public/app/core/services/bridge_srv.ts:87]: @ryantxu if you have time to test that this doesn't break the previous use case I would really appreciate it!

File: public/app/features/dashboard/components/RowOptions/RowOptionsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export const RowOptionsModal: FC<RowOptionsModalProps> = ({ repeat, title, onDismiss, onUpdate }) => {
+  return (
+    <Modal
+      isOpen={true}
+      title="Row Options"
+      icon="copy"
+      onDismiss={onDismiss}
+      className={css`
```
Comment (mckn) [inline public/app/features/dashboard/components/RowOptions/RowOptionsModal.tsx:21]: nit: in this case I would create a styles constant that holds the styling. Not that it is super important in this scenario but with this approach it makes the probability of continue to add styles in the same way lower.

**Review by mckn — APPROVED** [review]
LGTM! Awesome work as always 💪

File: public/app/features/dashboard/components/RowOptions/RowOptionsModal.tsx (hunk 3 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/RowOptions/RowOptionsModal.tsx:21]: Splendid catch and I agree!

**Review by ryantxu — APPROVED** [review]
@hugohaggmark -- everything seems to work ok :)  

github really needs to swap the green/red colors!
image

**Review by daniellee — CHANGES_REQUESTED** [review]
Seeing at least one bug in the Stackdriver query variable editor. The cases those tests are supposed to check are failing in the UI too.

When I create a new variable, the default value for query type is "projects". I would expect the preview to show a list of Stackdriver projects but even though `public/app/plugins/datasource/stackdriver/components/VariableQueryEditor.tsx` fetches a list of projects in componentDidMount, the preview is not being populated.

It does work if I switch to another query type and then back again when "this.props.onChange(queryModel, `Stackdriver - ${query.name}`);" is called.

It's not obvious to me how to fix this - could you give me a tour of the new variables code on Monday?

**Comment by hugohaggmark** [comment]
> @hugohaggmark -- everything seems to work ok :)
> 
> github really needs to swap the green/red colors!
> image

thanks @ryantxu this feels great!

**Comment by hugohaggmark** [comment]
> Seeing at least one bug in the Stackdriver query variable editor. The cases those tests are supposed to check are failing in the UI too.
> 
> When I create a new variable, the default value for query type is "projects". I would expect the preview to show a list of Stackdriver projects but even though `public/app/plugins/datasource/stackdriver/components/VariableQueryEditor.tsx` fetches a list of projects in componentDidMount, the preview is not being populated.
> 
> It does work if I switch to another query type and then back again when "this.props.onChange(queryModel, `Stackdriver - ${query.name}`);" is called.
> 
> It's not obvious to me how to fix this - could you give me a tour of the new variables code on Monday?

I'd love too help out, maybe we could solve this with pair programming?

File: public/app/core/services/bridge_srv.ts (hunk 2 above)
Comment (hugohaggmark) [reply public/app/core/services/bridge_srv.ts:87]: thanks for testing this out @ryantxu

**Review by dprokop — APPROVED** [review]
Great! Your GH stats are going to be high this week @hugohaggmark  ;)

**Review by daniellee — APPROVED** [review]
LGTM

Love the loading indicator in the Stackdriver variable dropdown - really nice touch.

**Comment by hugohaggmark** [comment]
> LGTM
> 
> Love the loading indicator in the Stackdriver variable dropdown - really nice touch.

Thank you @daniellee

**Comment by hugohaggmark** [comment]
@dprokop @torkelo when do you think is a good time to merge this?

**Comment by dprokop** [comment]
@hugohaggmark I think we should be good to merge it this week. This gives us 2 weeks before the beta to catch any problems. 

I think it also deserves some breaking change notice as I guess plugins may use `variablesSrv`, is that correct @ryantxu?

**Comment by hugohaggmark** [comment]
@dprokop I'll merge this after it turns green then 💥

File: public/app/features/templating/specs/template_srv.test.ts [hunk 4]
```diff
@@ -1,609 +0,0 @@
-import { TemplateSrv } from '../template_srv';
```
Comment (torkelo) [inline public/app/features/templating/specs/template_srv.test.ts:1]: this file was accidentally removed? contains super important tests for templating interpolation :)

**Pull request merged**